### PR TITLE
Implement WebSub Compiler Plugin

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes: [#1099](https://github.com/ballerina-platform/ballerina-standard-library/issues/1099)

## Checklist
- [X] Linked to an issue
- [X] Updated the changelog
- [X] Added tests

### CODE REVIEW

File: websub-compiler-plugin/src/main/java/io/ballerina/stdlib/websub/validator/ServiceDeclarationValidator.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+            Optional<ReturnTypeDescriptorNode> returnTypesOpt = functionDefinition.functionSignature().returnTypeDesc();
+            if (returnTypesOpt.isPresent()) {
+                ReturnTypeDescriptorNode returnTypeDescription = returnTypesOpt.get();
+                Node returnType = returnTypeDescription.type();
+                String returnTypeName = returnType.toString().trim();
+                List<String> availableReturnTypes = Arrays
+                        .stream(returnTypeName.replace(Constants.OPTIONAL, "").split("\\|"))
+                        .map(String::trim)
+                        .collect(Collectors.toList());
+                if (!predefinedReturnTypes.containsAll(availableReturnTypes)) {
```
Comment (chamil321) [inline websub-compiler-plugin/src/main/java/io/ballerina/stdlib/websub/validator/ServiceDeclarationValidator.java:231]: Rather doing a string match for return types, it's better of you use the sematic model and do proper type check. 
```
websub:Acknowledgement
```
can be stated with an alias
```
import ballerina/websub as hello;
hello:Acknowledgement
```

Comment (chamil321) [reply websub-compiler-plugin/src/main/java/io/ballerina/stdlib/websub/validator/ServiceDeclarationValidator.java:231]: `context.semanticModel().symbol(member).get()`

File: websub-compiler-plugin-test/build.gradle [hunk 2]
```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (MadhukaHarith92) [inline websub-compiler-plugin-test/build.gradle:2]:
```suggestion
 * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: websub-compiler-plugin-test/build.gradle [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    }
+}
+
+test {
+    useTestNG()
+}
+
+copyDistribution.dependsOn ":websub-ballerina:build"
+copyPackageBala.dependsOn copyDistribution
+test.dependsOn copyPackageBala
```
Comment (MadhukaHarith92) [inline websub-compiler-plugin-test/build.gradle:114]:
```suggestion
test.dependsOn copyPackageBala

```

**Comment by ayeshLK** [comment]
@shafreenAnfar / @ldclakmal / @chamil321 / @ThisaruGuruge / @Bhashinee / @aashikam please review this

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**